tests/test_trtllm_gen_attention.py

```diff
@@ -570,14 +617,37 @@ def test_trtllm_batch_decode(
             k_scale=k_scale,
             v_scale=v_scale / o_scale,
             enable_pdl=enable_pdl,
+            q_len_per_req=q_len_per_req,
         )
         # v_scale, o_scale in wrapper is emulated by multiplying output by v_scale instead of fused into kernel.
         if v_scale == o_scale == 1.0:
             assert (output_wrapper == output).all()
         else:
-            torch.testing.assert_close(
-                output.float(), output_wrapper.float(), rtol=1e-1, atol=1e-1
-            )
+            # todo(Yingyi): fix precision issue with this test
+            if not (
```

Comment:
I still have deep concern about the special handling of precision here, a more fundamental solution could be a PrecisionManager to centrialize the handling of tolerance for different data types.

Also, let's add a flashinfer.testing.assert_allclose() function where there could be arguments such as max mismatched elements, it will be fallback to torch.testing.assert_allclose when mismatched elements is not specified.